torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -242,8 +248,9 @@ void PropagateCatShape(Node * cat_node) {
     node->output()->setType(input_types[0]->withSizes(sizes));
```

Comment:
This is not correct now with `pad=True`.  If I understand the api correctly `sizes[i]` should be equal to the maximum over tp->size(i).